libs/geometry/src/polygon.rs

```diff
@@ -120,6 +121,20 @@ impl Polygon {
         let y = self.points.iter().map(|point| point.y).sum::<i64>() / self.points.len() as i64;
         Point::new(x, y)
     }
+
+    /// Helper function that checks if a point is contained within a triangle
+    fn traingle_contains(p: Point, v1: Point, v2: Point, v3: Point) -> bool {
+        let total_area = Polygon::triangle_area(v1, v2, v3);
+        let sum_area = Polygon::triangle_area(p, v2, v3)
+            + Polygon::triangle_area(v1, p, v3)
+            + Polygon::triangle_area(v1, v2, p);
+        sum_area == total_area
+    }
+
+    /// Helper function that finds the area of a given triangle
+    fn triangle_area(v1: Point, v2: Point, v3: Point) -> f32 {
```

Comment:
Move outside of `impl Polygon` and use integers/rationals instead of floats. Equality on floats is unreliable.